File: backend/src/controllers/locationController.js
const indiaLocations = require('../data/india-locations.json');

exports.getStates = (req, res) => {
  const states = indiaLocations.states.map((s) => ({ code: s.code, name: s.name }));
  res.json({ success: true, states });
};

exports.getDistricts = (req, res) => {
  const state = indiaLocations.states.find(
    (s) => s.code === req.params.stateCode || s.name === req.params.stateCode
  );
  if (!state) {
    return res.status(404).json({ success: false, message: 'State not found' });
  }
  res.json({ success: true, districts: state.districts });
};

exports.getCities = (req, res) => {
  const state = indiaLocations.states.find(
    (s) => s.code === req.params.stateCode || s.name === req.params.stateCode
  );
  if (!state) return res.status(404).json({ success: false, message: 'State not found' });

  const district = state.districts.find(
    (d) => d.name === req.params.districtName || d.code === req.params.districtName
  );
  if (!district) {
    return res.status(404).json({ success: false, message: 'District not found' });
  }
  res.json({ success: true, cities: district.cities || [] });
};

exports.getAllLocations = (req, res) => {
  res.json({ success: true, data: indiaLocations });
};

File: backend/src/main/java/com/pujasetu/controller/LocationController.java
package com.pujasetu.controller;

import com.pujasetu.service.LocationService;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/locations")
@RequiredArgsConstructor
@Tag(name = "Locations")
public class LocationController {

    private final LocationService locationService;

    @GetMapping("/states")
    public ResponseEntity<Map<String, Object>> getStates() {
        return ResponseEntity.ok(locationService.getStates());
    }

    @GetMapping("/states/{stateCode}/districts")
    public ResponseEntity<Map<String, Object>> getDistricts(@PathVariable String stateCode) {
        return ResponseEntity.ok(locationService.getDistricts(stateCode));
    }

    @GetMapping("/states/{stateCode}/districts/{districtName}/cities")
    public ResponseEntity<Map<String, Object>> getCities(
            @PathVariable String stateCode,
            @PathVariable String districtName
    ) {
        return ResponseEntity.ok(locationService.getCities(stateCode, districtName));
    }

    @GetMapping("/all")
    public ResponseEntity<Map<String, Object>> getAll() {
        return ResponseEntity.ok(locationService.getAllLocations());
    }
}
